web/client/utils/mapinfo/__tests__/wms-test.js

```diff
@@ -0,0 +1,163 @@
+            type: "wms",
+            name: "test_layer"
+        };
+        mockAxios.onGet().reply(() => {
+            return [200, {
+                "type": "FeatureCollection",
+                "features": [{}, {}],
+                "totalFeatures": "unknown",
+                "numberReturned": 2,
+                "timeStamp": "2023-09-22T08:50:30.808Z",
+                "crs": null
+            }];
+        });
+        getFeatureInfo(
+            "TEST_URL", {
+                info_format: INFO_FORMATS.PROPERTIES
+            }, SAMPLE_LAYER
+        ).subscribe(
+            n => {
+                expect(n.data.features.length).toEqual(2);
+                expect(n.features.length).toEqual(2);
+                done();
+            },
+            error => {
+                return done(error);
+            }
+        );
+    });
+    it('should return the response object from getIdentifyFlow in case no data features as a text,', (done) => {
+        const SAMPLE_LAYER = {
+            type: "wms",
+            name: "test_layer"
+        };
+        mockAxios.onGet().reply(() => {
+            return [200, 'no features were found'];
+        });
+        getFeatureInfo(
+            "TEST_URL", {
+                info_format: INFO_FORMATS.TEXT
+            }, SAMPLE_LAYER
+        ).subscribe(
+            n => {
+                expect(n.data).toEqual('no features were found');
+                done();
+            },
+            error => {
+                return done(error);
+            }
+        );
+    });
+    it('test intercept ogc error in wms if success', (done)=>{
+        mockAxios.onGet().reply(() => {
+            return [200,  {
+                "type": "FeatureCollection",
+                "features": [],
+                "totalFeatures": "unknown",
+                "numberReturned": 0,
+                "timeStamp": "2023-09-22T08:50:30.808Z",
+                "crs": null
+            }];
+        });
+        wms
+            .getIdentifyFlow(undefined, "/", { features: [] })
+            .subscribe((response) => {
+                expect(response?.data?.features).toEqual([]);
+                done();
+            });
+    });
+    it('test intercept ogc error in wms if exception', (done)=>{
+        mockAxios.onGet().reply(() => {
+            return [200, {
+                "version": "1.1.1",
+                "exceptions": [
+                    {
+                        "code": "LayerNotDefined",
+                        "locator": "layers",
+                        "text": "Could not find layer misc:Bonnebladen1900_index2121 Details:org.geoserver.platform.ServiceException: Could not find layer misc:Bonnebladen1900_index2121"
+                    }]
+            }];
+        });
+        wms
+            .getIdentifyFlow(undefined, "/", { features: [] })
+            .subscribe((response) => {
+                expect(response?.data?.exceptions).toExist();
+                done();
+            });
```

Comment:
are you sure this is actually testing the intercept ogc error call?

shouldn't throw an error and therefore the error callback should be called. so done have to placed there